apps/web/src/views/Swap/V3Swap/components/SwapModalFooter.tsx

```diff
@@ -25,30 +29,53 @@ export const SwapModalFooter = memo(function SwapModalFooter({
   lpFee: realizedLPFee,
   inputAmount,
   outputAmount,
+  trade,
   tradeType,
-  executionPrice,
   slippageAdjustedAmounts,
   isEnoughInputBalance,
   onConfirm,
   swapErrorMessage,
   disabledConfirm,
+  isMM,
+  isRFQReady,
+  currencyBalances,
 }: {
+  trade: SmartRouterTrade<TradeType> | undefined
   tradeType: TradeType
   lpFee: CurrencyAmount<Currency>
   inputAmount: CurrencyAmount<Currency>
   outputAmount: CurrencyAmount<Currency>
   priceImpact: Percent
-  executionPrice: Price<Currency, Currency>
   slippageAdjustedAmounts: { [field in Field]?: CurrencyAmount<Currency> }
   isEnoughInputBalance: boolean
-  onConfirm: () => void
   swapErrorMessage?: string | undefined
   disabledConfirm: boolean
+  isMM?: boolean
+  isRFQReady?: boolean
+  currencyBalances: {
+    INPUT?: CurrencyAmount<Currency>
+    OUTPUT?: CurrencyAmount<Currency>
+  }
+  onConfirm: () => void
 }) {
   const { t } = useTranslation()
   const [showInverted, setShowInverted] = useState<boolean>(false)
   const severity = warningSeverity(priceImpactWithoutFee)
 
+  const totalFeePercent = `${(TOTAL_FEE * 100).toFixed(2)}%`
+  const lpHoldersFeePercent = `${(LP_HOLDERS_FEE * 100).toFixed(2)}%`
+  const treasuryFeePercent = `${(TREASURY_FEE * 100).toFixed(4)}%`
+  const buyBackFeePercent = `${(BUYBACK_FEE * 100).toFixed(4)}%`
+
+  const executionPriceDisplay = useMemo(() => {
+    if (isMM) {
+      return mmFormatExecutionPrice(trade, showInverted)
+    }
+
+    const price = SmartRouter.getExecutionPrice(trade as SmartRouterTrade<TradeType>)
```

Comment:
Let's avoid using type assertion if possible. If we need null check then let's do the null check before calling the function